refactor: Extract functions to parent

rafaeelaudibert opened the pull request:
These functions will be reused in `stats_table.py`, so it makes sense to extract them one level higher. There's some extra complication on how we handle `self.query.compare` because `stats_table.py` always wants to compare, so we need that to be a parameter

This is just refactoring work to allow us to better filter by `ConversionGoal` on `stats_table.py`. If anything, this is just better code (to some definition of better)

rafaeelaudibert committed:
Extract functions to parent

These functions will be reused in `stats_table.py`, so it makes sense to extract them one level higher. There's some extra complication on how we handle `self.query.compare` because `stats_table.py` always wants to compare.

robbie-c commented:
`WebTopClicksQuery` can be deleted if doing so would make your life easier, it's not used any more

rafaeelaudibert commented:
> `WebTopClicksQuery` can be deleted if doing so would make your life easier, it's not used any more

@robbie-c It's not complicating me right now, but I'm always happy to remove code :)

robbie-c commented:
I was just looking at the mypy errors ;)

rafaeelaudibert commented:
> I was just looking at the mypy errors ;)

@robbie-c Oh, duh :). Might remove it then, thx

rafaeelaudibert committed:
Remove `WebTopClicksQuery`

This isnt being used anymore, and it's just complicating some of our code changes

rafaeelaudibert committed:
Force `Compare` to be passed explicitly

This is complicated because `self.query` doesn't always have `self.query.compare`, so this would fail a lot of tests

rafaeelaudibert committed:
Fix mypy errors

rafaeelaudibert commented on frontend/__snapshots__/scenes-other-onboarding--onboarding-billing--light.png:
Ahn, looks very wrong, hog is gone 🔥, I will make it regenerate this once Depot is back online

rafaeelaudibert committed:
Extract `compare` to child calls

rafaeelaudibert commented:
@robbie-c tests are passing now :)
